---
title: BrAPI Success Stories
keywords:
- brapi
- breeding
- api
- standards
- community
lang: en-US
date-meta: '2024-05-07'
author-meta:
- Peter "BrapMan" Selby
- Trevor "Cool Kid" Rife
- Khaled Al-Shamaa
- Isabelle Alic
- Sebastian "Baz" Raubach
- Paul Shaw
- Iain Milne
- Cyril Pommier
- Becky Smith
- Chaney Courtney
- Zachary Miller
- Terry Casstevens
- Sook Jung
- Taein Lee
- Josh "Big Data" Lamos-Sweeney
- Francisco López
- Marco Marsella
- Matthias Lange
- Guilhem Sempéré
- Stephan Weise
- Patrick König
- Manuel Feser
- Gouripriya Davuluri
- Paul Kersey
- Erwan Le-Floch
- Jospeh Ruff
- Michael Alaux
- Célia Michotey
- Anne-Francoise Adam-Blondon
- Jeremy Destin
- Maud Marty
- Suman Kumar
- Matthijs Brouwer
- Bert Droesbeke
- Jan Erik Backlund
- Aldrin Batac
- Mariano Crimi
- Corina Habito
- Nahuel Soldevilla
- Clarysabel Tovar
- Sebastian Beier
- Valentin Guignon
- Mathieu Rouard
- Asis Hallab
- Rafael Abbeloos
- Laszlo Lang
- Vivian Bass Vega
- Mirella Flores-Gonzalez
header-includes: |
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="BrAPI Success Stories" />
  <meta name="citation_title" content="BrAPI Success Stories" />
  <meta property="og:title" content="BrAPI Success Stories" />
  <meta property="twitter:title" content="BrAPI Success Stories" />
  <meta name="dc.date" content="2024-05-07" />
  <meta name="citation_publication_date" content="2024-05-07" />
  <meta property="article:published_time" content="2024-05-07" />
  <meta name="dc.modified" content="2024-05-07T20:03:29+00:00" />
  <meta property="article:modified_time" content="2024-05-07T20:03:29+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Peter &#34;BrapMan&#34; Selby" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-7151-4445" />
  <meta name="citation_author" content="Trevor &#34;Cool Kid&#34; Rife" />
  <meta name="citation_author_institution" content="Clemson University" />
  <meta name="citation_author_orcid" content="0000-0002-5974-6523" />
  <meta name="citation_author" content="Khaled Al-Shamaa" />
  <meta name="citation_author_institution" content="ICARDA" />
  <meta name="citation_author_orcid" content="0000-0002-7668-3798" />
  <meta name="citation_author" content="Isabelle Alic" />
  <meta name="citation_author_institution" content="INRAE" />
  <meta name="citation_author_orcid" content="0000-0002-8961-6068" />
  <meta name="citation_author" content="Sebastian &#34;Baz&#34; Raubach" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0001-5659-247X" />
  <meta name="citation_author" content="Paul Shaw" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-0202-1150" />
  <meta name="citation_author" content="Iain Milne" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-4126-0859" />
  <meta name="citation_author" content="Cyril Pommier" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-9040-8733" />
  <meta name="citation_author" content="Becky Smith" />
  <meta name="citation_author_institution" content="The James Hutton Institute" />
  <meta name="citation_author_orcid" content="0000-0002-8968-3383" />
  <meta name="citation_author" content="Chaney Courtney" />
  <meta name="citation_author_institution" content="Clemson University" />
  <meta name="citation_author" content="Zachary Miller" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0002-5454-4527" />
  <meta name="citation_author" content="Terry Casstevens" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author_orcid" content="0000-0001-7602-0487" />
  <meta name="citation_author" content="Sook Jung" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0003-3968-2769" />
  <meta name="citation_author" content="Taein Lee" />
  <meta name="citation_author_institution" content="Department of Horticulture, Washington State University" />
  <meta name="citation_author_orcid" content="0000-0002-2000-2586" />
  <meta name="citation_author" content="Josh &#34;Big Data&#34; Lamos-Sweeney" />
  <meta name="citation_author_institution" content="Cornell University" />
  <meta name="citation_author" content="Francisco López" />
  <meta name="citation_author_institution" content="International Treaty on Plant Genetic Resources for Food and Agriculture, FAO" />
  <meta name="citation_author" content="Marco Marsella" />
  <meta name="citation_author_institution" content="International Treaty on Plant Genetic Resources for Food and Agriculture, FAO" />
  <meta name="citation_author_orcid" content="0000-0003-0334-8785" />
  <meta name="citation_author" content="Matthias Lange" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0002-4316-078X" />
  <meta name="citation_author" content="Guilhem Sempéré" />
  <meta name="citation_author_institution" content="CIRAD (french agricultural research and international cooperation organization)" />
  <meta name="citation_author_institution" content="South Green Platform" />
  <meta name="citation_author_orcid" content="0000-0001-7429-2091" />
  <meta name="citation_author" content="Stephan Weise" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0003-4031-9131" />
  <meta name="citation_author" content="Patrick König" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_orcid" content="0000-0002-8948-6793" />
  <meta name="citation_author" content="Manuel Feser" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author_institution" content="Graduate School DILS, Bielefeld Institute for Bioinformatics Infrastructure (BIBI)" />
  <meta name="citation_author_orcid" content="0000-0001-6546-1818" />
  <meta name="citation_author" content="Gouripriya Davuluri" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author" content="Paul Kersey" />
  <meta name="citation_author_institution" content="Royal Botanic Gardens, Kew" />
  <meta name="citation_author" content="Erwan Le-Floch" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author" content="Jospeh Ruff" />
  <meta name="citation_author_institution" content="Royal Botanic Gardens, Kew" />
  <meta name="citation_author" content="Michael Alaux" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0001-9356-4072" />
  <meta name="citation_author" content="Célia Michotey" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0003-1877-1703" />
  <meta name="citation_author" content="Anne-Francoise Adam-Blondon" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author_orcid" content="0000-0002-3412-9086" />
  <meta name="citation_author" content="Jeremy Destin" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author" content="Maud Marty" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France" />
  <meta name="citation_author_institution" content="Université Paris-Saclay, INRAE, URGI, Versailles, France" />
  <meta name="citation_author" content="Suman Kumar" />
  <meta name="citation_author_institution" content="Leibniz Institute of Plant Genetics and Crop Plant Research" />
  <meta name="citation_author" content="Matthijs Brouwer" />
  <meta name="citation_author_institution" content="Wageningen University and Research" />
  <meta name="citation_author" content="Bert Droesbeke" />
  <meta name="citation_author_institution" content="VIB Data Core" />
  <meta name="citation_author_orcid" content="0000-0003-0522-5674" />
  <meta name="citation_author" content="Jan Erik Backlund" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Aldrin Batac" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Mariano Crimi" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Corina Habito" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author" content="Nahuel Soldevilla" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Clarysabel Tovar" />
  <meta name="citation_author_institution" content="Integrated Breeding Platform" />
  <meta name="citation_author_institution" content="Leafnode LLC" />
  <meta name="citation_author" content="Sebastian Beier" />
  <meta name="citation_author_institution" content="Institute of Bio- and Geosciences (IBG-4: Bioinformatics), CEPLAS, Forschungszenturm Jülich GmbH, Wilhelm Johnen Straße, 52428 Jülich, Germany" />
  <meta name="citation_author_institution" content="Bioeconomy Science Center (BioSC), Forschungszentrum Jülich GmbH, 52428 Jülich, Germany" />
  <meta name="citation_author_orcid" content="0000-0002-2177-8781" />
  <meta name="citation_author" content="Valentin Guignon" />
  <meta name="citation_author_institution" content="Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-0903-6811" />
  <meta name="citation_author" content="Mathieu Rouard" />
  <meta name="citation_author_institution" content="Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France" />
  <meta name="citation_author_orcid" content="0000-0003-0284-1885" />
  <meta name="citation_author" content="Asis Hallab" />
  <meta name="citation_author_institution" content="Jülich research center, Institute of Bio- and Geosciences (IBG), Bioinformatics (IBG-4) and Bingen Technical University of Applied Sciences, Germany" />
  <meta name="citation_author_orcid" content="0000-0002-2421-5431" />
  <meta name="citation_author" content="Rafael Abbeloos" />
  <meta name="citation_author_institution" content="VIB AgroIncubator" />
  <meta name="citation_author_orcid" content="0000-0002-0177-3887" />
  <meta name="citation_author" content="Laszlo Lang" />
  <meta name="citation_author_institution" content="Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany" />
  <meta name="citation_author_orcid" content="0009-0009-8936-4532" />
  <meta name="citation_author" content="Vivian Bass Vega" />
  <meta name="citation_author_institution" content="Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany" />
  <meta name="citation_author_orcid" content="0009-0002-2476-9888" />
  <meta name="citation_author" content="Mirella Flores-Gonzalez" />
  <meta name="citation_author_institution" content="Boyrce Thompson Institute" />
  <meta name="citation_author_orcid" content="0000-0002-7759-1617" />
  <link rel="canonical" href="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta property="og:url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta property="twitter:url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta name="citation_fulltext_html_url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/" />
  <meta name="citation_pdf_url" content="https://plantbreeding.github.io/BrAPI-Manuscript2/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://plantbreeding.github.io/BrAPI-Manuscript2/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://plantbreeding.github.io/BrAPI-Manuscript2/v/031169392c7b086850740141be23886c2e5a63cb/" />
  <meta name="manubot_html_url_versioned" content="https://plantbreeding.github.io/BrAPI-Manuscript2/v/031169392c7b086850740141be23886c2e5a63cb/" />
  <meta name="manubot_pdf_url_versioned" content="https://plantbreeding.github.io/BrAPI-Manuscript2/v/031169392c7b086850740141be23886c2e5a63cb/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
bibliography:
- content/manual-references.json
manubot-output-bibliography: output/references.json
manubot-output-citekeys: output/citations.tsv
manubot-requests-cache-path: ci/cache/requests-cache
manubot-clear-requests-cache: false
...






<small><em>
This manuscript
([permalink](https://plantbreeding.github.io/BrAPI-Manuscript2/v/031169392c7b086850740141be23886c2e5a63cb/))
was automatically generated
from [plantbreeding/BrAPI-Manuscript2@0311693](https://github.com/plantbreeding/BrAPI-Manuscript2/tree/031169392c7b086850740141be23886c2e5a63cb)
on May 7, 2024.
</em></small>



## Authors



+ **Peter "BrapMan" Selby**
  ^[✉](#correspondence)^<br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-7151-4445](https://orcid.org/0000-0001-7151-4445)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [BrapiCoordinatorSelby](https://github.com/BrapiCoordinatorSelby)
    <br>
  <small>
     Cornell University
     · Funded by NIFA-DSFAS 2022-67021-37024
  </small>

+ **Trevor "Cool Kid" Rife**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-5974-6523](https://orcid.org/0000-0002-5974-6523)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [trife](https://github.com/trife)
    <br>
  <small>
     Clemson University
  </small>

+ **Khaled Al-Shamaa**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-7668-3798](https://orcid.org/0000-0002-7668-3798)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [khaled-alshamaa](https://github.com/khaled-alshamaa)
    <br>
  <small>
     ICARDA
  </small>

+ **Isabelle Alic**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-8961-6068](https://orcid.org/0000-0002-8961-6068)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [Isabelle-inrae](https://github.com/Isabelle-inrae)
    <br>
  <small>
     INRAE
     · Funded by {'Short version': 'This work was supported by the project Phenome-ANR-11-INBS-0012'}; {'OR Long version': "This work was supported by the Agence Nationale de la Recherche, programme Investissements d'avenir, ANR-11-INBS-0012 (Phenome)"}
  </small>

+ **Sebastian "Baz" Raubach**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-5659-247X](https://orcid.org/0000-0001-5659-247X)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [sebastian-raubach](https://github.com/sebastian-raubach)
    <br>
  <small>
     The James Hutton Institute
  </small>

+ **Paul Shaw**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-0202-1150](https://orcid.org/0000-0002-0202-1150)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [cardinalb](https://github.com/cardinalb)
    <br>
  <small>
     The James Hutton Institute
  </small>

+ **Iain Milne**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-4126-0859](https://orcid.org/0000-0002-4126-0859)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [imilne](https://github.com/imilne)
    <br>
  <small>
     The James Hutton Institute
  </small>

+ **Cyril Pommier**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-9040-8733](https://orcid.org/0000-0002-9040-8733)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [cpommier](https://github.com/cpommier)
    <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
     · Funded by {'Short version': 'This work was supported by the project Phenome-ANR-11-INBS-0012'}; {'OR Long version': "This work was supported by the Agence Nationale de la Recherche, programme Investissements d'avenir, ANR-11-INBS-0012 (Phenome)"}
  </small>

+ **Becky Smith**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-8968-3383](https://orcid.org/0000-0002-8968-3383)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [Batbaby91](https://github.com/Batbaby91)
    <br>
  <small>
     The James Hutton Institute
  </small>

+ **Chaney Courtney**
  <br>
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [chaneylc](https://github.com/chaneylc)
    <br>
  <small>
     Clemson University
  </small>

+ **Zachary Miller**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-5454-4527](https://orcid.org/0000-0002-5454-4527)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [zrm22](https://github.com/zrm22)
    <br>
  <small>
     Cornell University
  </small>

+ **Terry Casstevens**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-7602-0487](https://orcid.org/0000-0001-7602-0487)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [tcasstevens](https://github.com/tcasstevens)
    <br>
  <small>
     Cornell University
  </small>

+ **Sook Jung**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-3968-2769](https://orcid.org/0000-0003-3968-2769)
    <br>
  <small>
     Department of Horticulture, Washington State University
  </small>

+ **Taein Lee**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-2000-2586](https://orcid.org/0000-0002-2000-2586)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [leetaei](https://github.com/leetaei)
    <br>
  <small>
     Department of Horticulture, Washington State University
  </small>

+ **Josh "Big Data" Lamos-Sweeney**
  <br>
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [jlamossweeney](https://github.com/jlamossweeney)
    <br>
  <small>
     Cornell University
  </small>

+ **Francisco López**
  <br>
  <small>
     International Treaty on Plant Genetic Resources for Food and Agriculture, FAO
  </small>

+ **Marco Marsella**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-0334-8785](https://orcid.org/0000-0003-0334-8785)
    <br>
  <small>
     International Treaty on Plant Genetic Resources for Food and Agriculture, FAO
  </small>

+ **Matthias Lange**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-4316-078X](https://orcid.org/0000-0002-4316-078X)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [langeipk](https://github.com/langeipk)
    <br>
  <small>
     Leibniz Institute of Plant Genetics and Crop Plant Research
  </small>

+ **Guilhem Sempéré**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-7429-2091](https://orcid.org/0000-0001-7429-2091)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [GuilhemSempere](https://github.com/GuilhemSempere)
    <br>
  <small>
     CIRAD (french agricultural research and international cooperation organization); South Green Platform
  </small>

+ **Stephan Weise**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-4031-9131](https://orcid.org/0000-0003-4031-9131)
    <br>
  <small>
     Leibniz Institute of Plant Genetics and Crop Plant Research
  </small>

+ **Patrick König**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-8948-6793](https://orcid.org/0000-0002-8948-6793)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [patrick-koenig](https://github.com/patrick-koenig)
    <br>
  <small>
     Leibniz Institute of Plant Genetics and Crop Plant Research
  </small>

+ **Manuel Feser**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-6546-1818](https://orcid.org/0000-0001-6546-1818)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [feserm](https://github.com/feserm)
    <br>
  <small>
     Leibniz Institute of Plant Genetics and Crop Plant Research; Graduate School DILS, Bielefeld Institute for Bioinformatics Infrastructure (BIBI)
  </small>

+ **Gouripriya Davuluri**
  <br>
  <small>
     Leibniz Institute of Plant Genetics and Crop Plant Research
  </small>

+ **Paul Kersey**
  <br>
  <small>
     Royal Botanic Gardens, Kew
  </small>

+ **Erwan Le-Floch**
  <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
  </small>

+ **Jospeh Ruff**
  <br>
  <small>
     Royal Botanic Gardens, Kew
  </small>

+ **Michael Alaux**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0001-9356-4072](https://orcid.org/0000-0001-9356-4072)
    <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
  </small>

+ **Célia Michotey**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-1877-1703](https://orcid.org/0000-0003-1877-1703)
    <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
  </small>

+ **Anne-Francoise Adam-Blondon**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-3412-9086](https://orcid.org/0000-0002-3412-9086)
    <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
  </small>

+ **Jeremy Destin**
  <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
  </small>

+ **Maud Marty**
  <br>
  <small>
     Université Paris-Saclay, INRAE, BioinfOmics, Plant Bioinformatics Facility, Versailles, France; Université Paris-Saclay, INRAE, URGI, Versailles, France
  </small>

+ **Suman Kumar**
  <br>
  <small>
     Leibniz Institute of Plant Genetics and Crop Plant Research
  </small>

+ **Matthijs Brouwer**
  <br>
  <small>
     Wageningen University and Research
  </small>

+ **Bert Droesbeke**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-0522-5674](https://orcid.org/0000-0003-0522-5674)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [bedroesb](https://github.com/bedroesb)
    <br>
  <small>
     VIB Data Core
  </small>

+ **Jan Erik Backlund**
  <br>
  <small>
     Integrated Breeding Platform
  </small>

+ **Aldrin Batac**
  <br>
  <small>
     Integrated Breeding Platform; Leafnode LLC
  </small>

+ **Mariano Crimi**
  <br>
  <small>
     Integrated Breeding Platform
  </small>

+ **Corina Habito**
  <br>
  <small>
     Integrated Breeding Platform
  </small>

+ **Nahuel Soldevilla**
  <br>
  <small>
     Integrated Breeding Platform; Leafnode LLC
  </small>

+ **Clarysabel Tovar**
  <br>
  <small>
     Integrated Breeding Platform; Leafnode LLC
  </small>

+ **Sebastian Beier**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-2177-8781](https://orcid.org/0000-0002-2177-8781)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [sebeier](https://github.com/sebeier)
    <br>
  <small>
     Institute of Bio- and Geosciences (IBG-4: Bioinformatics), CEPLAS, Forschungszenturm Jülich GmbH, Wilhelm Johnen Straße, 52428 Jülich, Germany; Bioeconomy Science Center (BioSC), Forschungszentrum Jülich GmbH, 52428 Jülich, Germany
  </small>

+ **Valentin Guignon**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-0903-6811](https://orcid.org/0000-0003-0903-6811)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [guignonv](https://github.com/guignonv)
    <br>
  <small>
     Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France
  </small>

+ **Mathieu Rouard**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0003-0284-1885](https://orcid.org/0000-0003-0284-1885)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [mrouard](https://github.com/mrouard)
    <br>
  <small>
     Bioversity International, Parc Scientifique Agropolis II, 34397 Montpellier, France
  </small>

+ **Asis Hallab**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-2421-5431](https://orcid.org/0000-0002-2421-5431)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [https://github.com/asishallab](https://github.com/https://github.com/asishallab)
    <br>
  <small>
     Jülich research center, Institute of Bio- and Geosciences (IBG), Bioinformatics (IBG-4) and Bingen Technical University of Applied Sciences, Germany
  </small>

+ **Rafael Abbeloos**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-0177-3887](https://orcid.org/0000-0002-0177-3887)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [raabb](https://github.com/raabb)
    <br>
  <small>
     VIB AgroIncubator
  </small>

+ **Laszlo Lang**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0009-0009-8936-4532](https://orcid.org/0009-0009-8936-4532)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [LzLang](https://github.com/LzLang)
    <br>
  <small>
     Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany
  </small>

+ **Vivian Bass Vega**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0009-0002-2476-9888](https://orcid.org/0009-0002-2476-9888)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [VivianBass](https://github.com/VivianBass)
    <br>
  <small>
     Bingen Technical University of Applied Sciences, Berlinstraße 109, 55411 Bingen am Rhein, Germany
  </small>

+ **Mirella Flores-Gonzalez**
  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon width=16 height=16}
    [0000-0002-7759-1617](https://orcid.org/0000-0002-7759-1617)
    · ![GitHub icon](images/github.svg){.inline_icon width=16 height=16}
    [mflores2021](https://github.com/mflores2021)
    <br>
  <small>
     Boyrce Thompson Institute
  </small>


::: {#correspondence}
✉ — Correspondence possible via [GitHub Issues](https://github.com/plantbreeding/BrAPI-Manuscript2/issues)
or email to
Peter "BrapMan" Selby \<ps664@cornell.edu\>.


:::


## Abstract {.page_break_before}

Population growth and climate change require extraordinary efforts to increase efficiency in breeding programs around the world. In the last few years, genomics technologies and genomic prediction approaches have provided a boost in genetic gain in breeding, but has also created a flood of data that needs careful management to be fully harnessed.
The Breeding API (BrAPI) project is an international, grass-roots effort to enable more efficient data management by enabling interoperability among plant breeding databases and tools, using a standardized RESTful web service API specification for communicating plant breeding data.
This community driven standard is free to be used by anyone interested in plant breeding data management, including trial data, phenotypic data and genotyping data management.
This manuscript describes advances in implementations of BrAPI in different breeding tools, and outlook for the current version of BrAPI. 


## Introduction

<!-- 
Notes:
* DIGITAL ECOSYSTEM
* Justification for a second paper
* Highlight Community Growth
* Hackathons
* sabbaticals
* Updates for V2
* Auth Data Access Control - super simplified - highlight data privacy and security elements
* Project structure - Modules
* Standardization of endpoints across objects - said in a non-tech way
-->

To address consequences of climate change and population growth, plant and animal breeding need to become more efficient and data driven to produce bigger, better, healthier, more sustainable crops. Modern breeding techniques require large amounts of high quality data to be effective, requiring digital methods for data collection, management, and analysis. Interoperability between breeding software tools, systems, and databases can substantially increase the efficiency of a breeding program. The ability to efficiently share data means access to larger and more complete datasets, enabling to build more accurate computational models and produce more accurate predictions and improved selections.

Plant and animal breeding is an incredibly important part of today's society. Almost every country in the world has some kind of breeding program supporting the agricultural community to produce bigger, better, healthier, and more sustainable crops. Climate change, population growth, disease mitigation, and nutritional health are all major challenges facing humanity, and the ability to breed better crops is a critical part of the solution. Modern breeding techniques require large amounts of high quality data to be effective. In the digital age, that breeding data is being collected, managed, and analyzed with computer software. Interoperability between breeding software tools, systems, and databases can substantially increase the efficiency of a breeding program. The ability to share tools gives each program a boost in computational power. The ability to share data allows the community to leverage access to larger, more complete, datasets; similarly, the community can build more accurate computational models and produce more accurate predictions.

The Breeding API (BrAPI) project is an effort to enable interoperability among breeding tools, systems, and databases. BrAPI is a standardized Representational State Transfer (REST), web service, Application Programming Interface (API), specification for breeding and related agricultural data. [@doi:10.1093/bioinformatics/btz190] By using the BrAPI standard, breeding software can more easily become interoperable, allowing groups to more easily share data and software tools. 

This manuscript will present BrAPI in its latest iteration. This includes a technical description of the standard and a showcase of the applications, services, and tools available in the BrAPI community. BrAPI has become an essential part of the digital infrastructure for breeding applications and related agricultural projects. It is the intention of this manuscript to demonstrate the value of BrAPI to the wider scientific community as an effective and efficient means to collaborate and share resources. 

### How it works

An Application Programming Interface (API) is a technical connection between two pieces of software. Just as a Graphical User Interface (GUI) or a Command Line Interface (CLI) allows a human user to interact with a piece of software, an API allows one software application to interact with another. A GUI or CLI might allow a user to input data, read data, and start processes within an application. An API allows one piece of software (sometimes called a client, user agent, or service consumer) to programmatically input data, read data, and start process within another piece of software (sometimes called a server or service provider).

A Representational State Transfer (REST) web service is a type of API commonly used in today's modern web infrastructure. REST is a technical architecture that describes the stateless transmission of data between applications. Typically, REST systems are implemented using the standard HTTP protocol that most of the modern internet is built upon. REST implementations also generally use JavaScript Object Notation (JSON) to represent the data being transferred. Both HTTP and JSON are programming language agnostic, very stable, and very flexible. This means BrAPI can be implemented in almost any piece of software, and can solve a wide range of use cases.

Data repositories and service providers can choose to represent their data as a BrAPI compatible API. By mapping the internal data structures to the standard models, data repositories can easily expose data to the outside world. Similarly, they can accept new data from external sources and automatically map the new data into the existing database. Client application developers can take advantage of this standardization by building tools that can easily integrate with all other BrAPI compatible data repositories. Visualization, reporting, analytics, data collection, and quality control tools can be built once and shared with other organizations following the standards. As the number of BrAPI compatible databases, tools, and organizations grows, so does the value added by implementing the standard into a given application. 

### Project Updates

Over its lifetime, the BrAPI project has grown and changed substantially. The latest stable version of the specification (v2.1) looks vastly different from the original version (v1.0) released in 2017. The total size of the specification has almost quadrupled in that time, going from 51 endpoints documented in v1.0 to 201 endpoints documented in v2.1. Because of this growth, the specification documents were reorganized into four modules: BrAPI-Core, BrAPI-Germplasm, BrAPI-Genotyping, and BrAPI-Phenotyping. Figure {@fig:domains} shows a simplified domain map of the whole BrAPI v2.1 data model, divided into the organizational modules. The early versions of the specification focused on read-only phenotype data, with a small consideration to the other domains. Now the specification has a full representation of most of the major concepts applicable to the breeding process. The new specification is also internally consistent, easier to navigate, and allows for read, write, and update capabilities. None of those qualities were a guarantee for the earlier versions. 

![A simplified domain map of the whole BrAPI data model, divided into organizational modules. A more detailed Entity Relationship Diagram (ERD) is available on brapi.org.](images/BrAPI_Domains_v2-1_vertical.png){#fig:domains width="100%"}

As the specification has matured, so have the tools, services, and libraries available to the community to work with the specification. Every version of the specification is now released with a change log to guide developers upgrading from a previous version, an Entity Relationship Diagram (ERD) to describe the whole data model visually, and a JSON Schema version of the model to be used in some automated development efforts. For groups who are using Java, Java Script, Python, R, or Drupal, there are community maintained libraries available that contain full BrAPI implementations ready to be added to some existing code. The BrAPI Test Server and the BRAVA validation tool are both still available to the community for testing purposes, and they have been maintained to support every version of the specification. Finally, the three new resources list pages on brapi.org advertise the other BrAPI compatible software available in the community. The BrAPPs list displays the 10 standalone, plug-and-play, applications available to the community. The servers list displays the 27 registered public data servers, their current status, and a form for registering additional servers. The compatible software list shows the 31 software applications that are BrAPI compliant, and again, a form for registering additional applications. The process for registering new items for these lists is completely voluntary, so the totals represent a lower bound for the number of BrAPPs, data repositories, and applications available in the the community. 

### Community Growth

The international BrAPI Community consists of software developers, breeders, and related scientists working on BrAPI related projects and data sources. This community is what sustains the BrAPI project, building implementations, maintaining development tools, and providing input to enhance the specification. As the project has grown, so has the community. The BrAPI project started in June 2014 with less than ten people coming together to discuss the idea. Over the next nine years, the community has grown to between 200 and 250 members. The community mailing list has 208 members, and the BrAPI Slack workspace has 234 members. The project leadership uses the mailing list to broadcast newsletters, announcements, and updates to the community. The BrAPI Slack workspace allows members of the community to discuss specific topics and collaborate directly with each other. 

The BrAPI Hackathons are a major staple of the BrAPI community. Twice a year, the community gathers to discuss the specification and collaborate on BrAPI related projects. This time is very valuable to the community; for some organizations, the hackathon is the only time during the year when they have time to work on anything related to BrAPI. During the COVID-19 pandemic, virtual hackathons took the place of in person events. While the virtual hackathons do not provide the same level of face-to-face time that is crucial to collaborative work, they did allow for more attendees to gather and share their opinions. The typical virtual hackathon has about twice as many registered attendees compared to an in-person hackathon. However, attendees have reported much more productive work time during the in-person events. As a compromise, going forward, the community leadership has decided to have one in-person hackathon and one virtual hackathon each year, to balance the advantages of both. 

As the project has matured, a formal project leadership structure became increasingly important. As described in Figure {@fig:org-chart}, the project governance is divided into two groups. The Project Management team is responsible for the day to day operations of the project. The PI and Co-PI are responsible for the project funds, and responsible for hiring the BrAPI Project Coordinator who is paid by the project funds. The Advisory Board is a group of elected officials representing the community. The board is responsible for long term planning of the project, as well as quick decision making on behalf of the community. The two groups meet quarterly to report on progress and stay synchronized. 

![The formal governance structure of the BrAPI Project Community](images/BrAPI_org_structure.jpg){#fig:org-chart width="100%"}


## Success Stories

<!-- success stories highlighting BrAPI usefulness in breeding cycle. Perhaps reference the original BrAPI paper where possible use cases were proposed.  -->

Below are a number of short success stories from the BrAPI community. These tools, applications, and infrastructure projects serve as another indicator of community growth and success over the past 5-10 years. These stories clearly illustrate all the different ways the BrAPI Standard can be used productively and in practice. 


### Data Collection 
<!-- (Field Book, Climmob, other Phenoapps, ImageBreed, etc) -->

<!-- * General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->


#### Field Book

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
Phenotypic data collection is an essential part of the breeding process. Historically, gathering data in the field was done with pen and paper, or perhaps some version of a digital spreadsheet. The abundance and prevalence of smart phones has allowed the Field Book mobile app to enhance data collection. Field Book can create well-formed digital observation records from the moment they are taken. This can improve the efficiency of data collection and reduce human error. 

In 2018, BrAPI was introduced into Field Book; specifically, the Core and Phenotyping modules. BrAPI was able to take things a step further by automating the flow of data from the Field Book mobile app to a central database server. This workflow allows data collection and storage to be expedited, removing the need of the user to transfer export files manually. Since Field Book’s adoption of BrAPI, many community servers have been integrated to simplify data storage. In this work flow, data is collected and stored completely digitally with little-to-no human involvement.


#### ClimMob

Not all data can be collected by a single person, or even by a single organization. ClimMob is a tool to easily allow citizen scientists to assist in the data collection process. Although this data may not be as detailed as a focused scientific program, it can be very useful to collect simple data from a wide range of locations and environments.

When it comes to BrAPI compatibility, ClimMob follows the same patterns established by Field Book. During a survey, all the farmer collected data is stored in a central ClimMob node. When the survey is complete, all the data is uploaded automatically via BrAPI to a central breeding database for long term storage and analysis. 


#### ImageBreed

High-throughput phenotyping has been gaining significant traction lately as a way to collect lots of data very quickly. Image collection from unmanned arial and ground vehicles (UAVs and UGVs) are a great way to collect a lot of raw data all at once, then analyze it later. ImageBreed is a image collection pipeline tool to support regular use of UAVs and UGVs. 

When the raw images have been processed through the standardization pipelines in ImageBreed, useful phenotypes can be extracted from the images. The BrAPI standard is used to push these phenotypes back to a central breeding database where they can be analyzed with other data. In addition to this, ImageBreed also has the option to use BrAPI to upload the raw images to the central breeding database, or any other BrAPI compatible long term storage service. The BrAPI models in the current version of the standard (V2.1) are rudimentary, but effective. The ImageBreed team has put in some work to enhance the BrAPI image data standards. 


#### GridScore

Phenotypic data collection underpins scientific crop research and plant breeding. Knowledge gained from collected data and its analysis alongside data visualizations inform further phenotypic trials and ideally support research hypotheses. The importance of accuracy and efficiency in the collection of this data as well as the infrastructure to facilitate the flow of data from the field to a knowledge base cannot be underestimated. [GridScore](https://ics.hutton.ac.uk/get-gridscore/) [@doi:10.1186/s12859-022-04755-2] is a modern mobile application for phenotypic observations that harnesses technological advancements in the area of mobile devices to enrich the data collection process.

BrAPI has further increased the value of GridScore by integrating it into the overarching workflow from trial creation, data collection, and its ultimate data storage for further processing. Specifically, trial designs as well as trait definitions can be imported into GridScore using BrAPI and a finalized trial can ultimately be exported via BrAPI to any compatible database.


### Data Management 

<!-- * General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->


#### PHIS 

The Hybrid Phenotyping Information System ([PHIS](http://www.phis.inrae.fr/) [@doi:https://doi.org/10.1111/nph.15385]), based on the [OpenSILEX](https://github.com/OpenSILEX/) framework, is an ontology-driven information system based on semantic web technologies. PHIS is deployed in several field and greenhouse platforms of the national [PHENOME](https://www.phenome-emphasis.fr/) and European [EMPHASIS](https://emphasis.plant-phenotyping.eu/) infrastructure. It manages and collects data from Phenotyping and High Throughput Phenotyping experiments on a day to day basis. PHIS unambiguously identifies all the objects and traits in an experiment, and establishes their types and relationships via ontologies and semantics.

PHIS has been designed to be BrAPI-compliant. PHIS adheres to the standards and protocols specified by BrAPI and implements various services aligning with the BrAPI standards, encompassing the Core, Phenotyping, and Germplasm modules. This enables integration and compatibility with BrAPI-compliant systems and platforms. This prerequisite served as the basis for formalizing the data model, while also facilitating compatibility with other standards, such as the Minimal Information About a Plant Phenotyping Experiment ([MIAPPE](https://www.miappe.org/) [@doi:https://doi.org/10.1111/nph.16544]). By integrating BrAPI requirements into its structure, PHIS not only meets the standards of the phenotyping field, but also strengthens its capacity for interoperability and effective collaboration in the wider context of plant breeding and related fields.

The fact that data within a PHIS instance can be queried through BrAPI services makes the indexing of PHIS in [FAIDARE](https://urgi.versailles.inra.fr/faidare/) very easy to implement.

Furthermore, as PHIS offers BrAPI-compliant Web Services, it simplifies the integration and data exchange with other European information systems that handle phenotyping data. The adherence to BrAPI standards ensures a common interface and compatibility, facilitating communication and collaboration between PHIS and other systems in the European context. This interoperability not only eases data sharing, but also promotes a more coherent and efficient approach to the management and use of phenotyping data on various platforms and research initiatives within the European scientific community.


#### DeltaBreed

<!-- Shawn Y.-->
DeltaBreed is an open-source data management system designed and developed by Breeding Insight to support USDA-ARS specialty crop and animal breeders. DeltaBreed is a unified system for managing breeding data that connects a variety of BrAPI applications (see list below). BrAPI integration allows the complexity underlying interoperability to be hidden, shielding users from multifactorial differences between diverse applications. DeltaBreed, adhering to the BrAPI model, establishes data standards and validations for users and provides a singular framework for data management and user training.

DeltaBreed users need not be aware of BrAPI or the specifics of underlying applications but will notice that BrAPI interoperability reduces the need for human-mediated file transfers and data manipulation. Field Book users, for example, can connect to their DeltaBreed program, authenticate, and pull studies and traits directly from DeltaBreed to Field Book on their data collection device. The subsequent step of pushing observations from Field Book to DeltaBreed is straightforward via BrAPI, but will not be implemented until repeated observation handling workflows are established to differentiate and validate repeated observations, such as accidental repeats, overwrite requests, time-series observations, and repeated sub-entity measures. Users can expect DeltaBreed observation handling to become more seamless with future development.

**DeltaBreed Connected Applications**
<< Submission is expected April 2024. We may need to trim this aspirational list down to reality in final edits.>>

+ BIMS <https://www.breedwithbims.org/>
+ BrAPI Java Server <https://test-server.brapi.org/brapi/v2/>
+ BrAPI Sync <https://github.com/IntegratedBreedingPlatform/brapi-sync>
+ BreedBase <https://breedbase.org/>
+ Diversity Arrays Technologies (DArT) genotyping services
+ Field Book <https://play.google.com/store/apps/details?id=com.fieldbook.tracker>
+ Gigwa <https://gigwa.southgreen.fr/gigwa/>
+ Mr Bean <https://github.com/AparicioJohan/MrBeanApp>
+ Pedigree Viewer <https://github.com/solgenomics/BrAPI-Pedigree-Viewer>


#### BMS

The [Breeding Management System (BMS)](https://bmspro.io), developed by the [Integrated Breeding Platform (IBP)](https://integratedbreeding.net/), is a suite of tools designed to enhance the efficiency and effectiveness of plant breeding. BMS covers all stages of the breeding process, with the emphasis on germplasm management and [ontology](https://cropontology.org)-harmonized phenotyping. It also features analytics and decision-support tools. With its focus on interoperability, BMS integrates smoothly with BrAPI, facilitating easy connections with a broad array of complementary tools and databases, notably [Gigwa](https://southgreen.fr/content/gigwa) which is deployed together with the BMS to fulfill the genotyping data management needs of BMS users.

The [brapi-sync](https://github.com/IntegratedBreedingPlatform/brapi-sync) tool, a significant component of BMS’s BrAPI capabilities, was developed by the IBP and released as a BrAPP for community use. Brapi-sync is designed to enhance collaboration among partner institutes within a network such as Innovation and Plant Breeding in West Africa ([IAVAO](https://www.iavao.org/en)), by enabling the sharing of germplasm and trials across BrAPI-enabled systems. This tool helps overcome traditional barriers to collaboration, ensuring data that was once isolated within specific programs or platforms can now be easily shared, integrated, and synchronized. 

Additionally, brapi-sync improves data management by utilizing the externalReferences field to maintain links to the origin IDs of each entity it transmits. This not only retains the original context of the data but also establishes a traceability mechanism for accurate data source attribution and verification. Such practices are crucial for maintaining data integrity and fostering trust among collaborative partners, ensuring access to accurate, reliable, and current information.


#### Breedbase

Breedbase is a comprehensive breeding data management system [@doi:10.1093/g3journal/jkac078] [@doi:10.1371/journal.pone.0240059] that implements a digital ecosystem for all breeding data, including trial data, phenotypic data, and genotypic data. Data acquisition is through tabled-based apps such as Fieldbook [@doi:10.2135/cropsci2013.08.0579] and related apps, such as Coordinate and InterCross apps, through drone imagery, Near Infra-Red Spectroscopy (NIRS), and other technologies. Search functions such as the Search Wizard interface provide powerful query capabilities, and various breeding-centric analysis tools are available, including mixed models, heritability, stability, PCA, and various clustering algorithms. The original impetus for creating Breedbase was the advent of new breeding paradigms based on genomic information such as genomic prediction algorithms [@doi:10.1093/genetics/157.4.1819] and the accompanying data management challenges, and complete genomic prediction workflow is integrated in the system.  The first instance was created for the NextGen Cassava project in 2012 as the Cassavabase (<https://cassavabase.org/>) database. Databases for other CGIAR root, tuber and banana (RTB) crops followed with database for yam (<https://yambase.org/>), sweet potato (<https://sweetpotatobase.org/>), banana (<https://musabase.org/>) as well as instances in labs and companies. The BrAPI interface [@doi:10.1093/bioinformatics/btz190] is crucial for Breedbase: Breedbase communicates via BrAPI with the data collection tablets, connection to other projects such as CLIMMOB [@doi:10.1016/j.compag.2023.108539], and many native tools use the BrAPI interface for accessing data. Users also appreciate the ability to connect to Breedbase instances using packages such as QBMS <https://icarda-git.github.io/QBMS/> for data import into R for custom analyses. Breedbase has been an early and continuous adopter of, and contributor to, the BrAPI standard.


#### BIMS

BIMS (Breeding Information Management System) [@doi:10.1093/database/baab054] is a free, secure, and online breeding management system which allows breeders to store, manage, archive, and analyze their private breeding program data. BIMS enables individual breeders to have complete control of their own breeding data along with access to tools such as data import/export, data analysis and data archiving for their germplasm, phenotype, genotype, and image data. BIMS is currently implemented in five community databases, the Genome Database for Rosaceae [@doi:10.1093/nar/gky1000], CottonGEN [@doi:10.3390/plants10122805], the Citrus Genome Database, the Pulse Crop Database, and the Genome Database for Vaccinium, as well as a crop-independent website, <https://breedwithbims.org>. BIMS in these five community databases enables individual breeders to import publicly available data so that they can utilize public data in their breeding program. BIMS utilizes the Android App Field Book, enabling seamless data transfer between BIMS and the Field Book App through either files or BrAPI. Data transfer through BrAPI between BIMS and other resources such as BreedBase, GIGWA, and Breeder Genomics Hub is also on the way. 


#### Germinate

[Germinate](https://ics.hutton.ac.uk/get-germinate/) [@doi:10.1002/csc2.20248] is an open-source plant genetic resources database that combines and integrates various kinds of plant breeding data including genotypic data, phenotypic trials data, passport data, images, geographic information and climate data into a single repository. Germinate is tightly linked to the BrAPI specification and supports a majority of BrAPI endpoints for querying, filtering and submission.

Germinate integrates and connects with other BrAPI-enabled tools such as GridScore for phenotypic data collection, Flapjack for genotypic data visualization and Helium for pedigree visualization, but, due to the nature of BrAPI, Germinate can act as a data repository for any BrAPI-compatible tool. Thanks to the interoperability provided by BrAPI the need for manual data handling becomes a rarity with the direct benefit of faster data processing, fewer to no human errors, data security and integrity.


#### PIPPA

[PIPPA](https://pippa.psb.ugent.be) is a data management system used for collecting data from the [WIWAM](https://www.wiwam.be/) range of automated high throughput phenotyping platforms. These platforms have been deployed at different research institutes and commercial breeders across Europe in a variety of configurations with different types of equipment such as weighing scales, cameras and environment sensors. Examples are:

+ [Umea Plant Science Centre](https://www.upsc.se/plant-growth-facilities-at-upsc-and-slu-umea/325-upsc-tree-phenotyping-platform.html)
+ [Fondazione Edmund Mach](https://cri.fmach.it/en/Facilities/Technological-Facilities/Plant-Phenotyping#application_fields)
+ [Phenovision](https://www.psb.ugent.be/phenotyping/phenovision)

Developed from 2016 onwards, the software features a web interface with functionality for setting up new experiments for the platform(s), planning imaging and irrigation treatments, linking metadata to pots (genotype, growth media, manual treatments), exporting data, importing data and visualizing data as charts. It also supports the integration of image analysis scripts and connections to a compute cluster for job submission.

To share the phenotype data of the experiments linked to publications, an implementation of BrAPI 1.3 was developed on a separate public PIPPA server open to the public, which allowed read only access to the data in a standardized format. This endpoint was registered on [FAIDARE](https://urgi.versailles.inra.fr/faidare/) and allows the data to be found alongside data from other BrAPI endpoints.

As the BrAPI ecosystem has matured, it created a clear path for the development of PIPPA as to how to share data in a manner according to the FAIR principles which are becoming standard in plant research data management best practices. In combination with the support for [MIAPPE](https://www.miappe.org/), these have served as guidelines in the current development, which is focussed on delivering a public BraPI 2.1 endpoint and making more high throughput datasets publicly available via BrAPI.


#### MGIS

<!-- authors: Mathieu Rouard, Valentin Guignon -->
The Musa Germplasm information system, [MGIS](https://www.crop-diversity.org/mgis/), serves as a comprehensive community portal dedicated to banana diversity, a crop critical to global food security [@doi:10.1093/database/bax046]. MGIS offers detailed information on banana germplasm, focusing on the collections held by the CGIAR International Banana Genebank (ITC) [@doi:10.1186/s43170-020-00015-6]. It is built on the Build on the Drupal/Tripal technology, like BIMS and Florilège.  Since its inception, MGIS developers have actively participated in the Breeding API (BrAPI) community, pushing for the integration of Multicrop Passport Data (MCPD) into Germplasm module call of the API. MGIS thus provides passport data information on ITC banana genebank accessions (with GLIS DOI), synchronized with [Genesys](https://www.genesys-pgr.org/a/overview/v2YdWZGrZjD), but also enriches it by incorporating additional data from other germplasm collections worldwide. All those germplasm data are available through BrAPI germplasm module calls implementations. For genotyping data, MGIS incorporates GIGWA [@doi:10.1093/gigascience/giz051], which provides tailored implementations for BrAPI genotyping module calls. Furthermore, MGIS supports the implementation of a set of BrAPI phenotyping module calls, facilitating the exposing of morphological descriptors and trait information supported by ontologies like the Crop Ontology [@doi:10.1093/aobpla/plq008]. It is integrated with the Trait Selector BrAPP, developed as part of a project involving Breedbase [@doi:10.1093/g3journal/jkac078]. Uses cases between the Musa implementation of Breedbase, MusaBase, and MGIS to interlink genebank and breeding data.


### Federated Data Management Infrastructures
<!-- 
(AGENT, INCREASING, EURISCO, DataPLANT, NFDI4BioDiversity, FAIRAgro)

* General use case description(s) -  AgrosystemIntegration of germplasm collections in context of data trustee models among private economy and public research, integration of ex-situ genebanks (EU H2020 projects AGENT, INCREASING), integrated agrosystems and plant research infrastructure
* Specific tool examples - BraPI endpoints for AGENT, IPK-Genebank, MIAPPE ISA-TAB2BRAPI service
* Alternate solutions/ why is it better with BrAPI - Schema.ORG lightweight meta data harvesting, ARCs as collaborative data decoration, API and publication pipeline
* future related use cases, areas to improve - LIMS to BrAPI proxies -->


<!-- Contribution BrAPI 2.0 paper
Suggested Authors: Matthias Lange, Patrick König, Stephan Weise, Gouripriya Davuluri, Suman Kumar, Joseph Ruff, Paul Kersey, Cyril Pommier, Michael Alaux, Erwan Le-Floch -->
#### AGENT Portal

In the global system for ex situ conservation of plant genetic resources (PGR) [@doi:10.3390/plants10081557], a total of ~5.8 million accessions are conserved in 1750 ex situ genebanks [@doi:10.2135/cropsci2017.01.0014]. Unique and permanent identifiers in the form of DOIs are available for more than 1.7 million accessions [@doi:Food and Agriculture Organization (FAO) The Global Information System for PGRFA]. Each DOI is linked to some basic descriptive data that facilitates the use of these resources. Many DOIs are also linked to additional data from different domains or will be in the future. In order to answer questions on the global biological diversity of a plant species, on duplicate detection, on provenance tracking for the identification of genetic integrity, on the selection of the most suitable material for various purposes, including breeding and research, and to support further applications in data mining or AI, a data space beyond the most basic information is needed that includes genotypic and phenotypic data. In this context, the aim of the AGENT project (<https://www.agent-project.eu/>) funded by the European Commission is to develop a concept for the digital exploitation and activation of this GenRes data space via European ex situ genebanks according to the FAIR criteria [@doi:10.1038/sdata.2016.18] and to test it in practice using two important crops, barley and wheat. In two work packages, standards and technology for data interoperability will be developed to establish a genetic resources infrastructure, which regulates data acquisition of genotypic and phenotypic data, integrates and archives them and makes them accessible according to FAIR principles. To this end, 13 European genebanks and 5 bioinformatics centers are cooperating and have agreed on standards and protocols for (i) the data flow (see figure {@fig:AGENT_Genotyping_Data_Flow}) and data formats [@doi:10.12688/f1000research.109080.2] for central archiving of genotypic and phenotypic data.

![Figure Data flow of genotypic data from AGENT partner databases](images/AGENT_Genotyping_Data_Flow.png){#fig:AGENT_Genotyping_Data_Flow width="100%"}

The AGENT portal as described in more detail in section unlock the full potential of the biological material stored in genebanks around the globe by using FAIR international data standards and an open digital infrastructure for the management of plant genetic resources. The implemented BrAPI interface enables to mine current and historic genotypic and phenotypic information to drive the discovery of genes, traits and knowledge for future missions, complement existing information for wheat and barley and the new data standards and infrastructure to foster an improved management of PGR for other crop species across European genebanks.

For the joint research data infrastructure for the federation of collections of genotypic and phenotypic data from European gene banks and bioinformatics institutes, a AGENT portal ({@fig:AGENT_WebFrontend}) as database infrastructure for integrated plant genetic resources on ex-situ genebanks is being created. It provides, manual data exploration, machine-readable access via BrAPI and provide data to the cored data deposition resources at the European Bioinformatics Institute (EBI).

![Figure AGENT Portal](images/AGENT_WebFrontend.png){#fig:AGENT_WebFrontend width="100%"}

The AGENT database backend aggregates curated and integrated passport data, phenotypic and genotypic data about wheat and barley accessions of 18 project partners are harmonized and integrated via BrAPI endpoints (<https://github.com/AGENTproject/BrAPI>) and explorable in a web portal (<https://agent.ipk-gatersleben.de>). The BrAPI endpoints were made available by scattered implementation. Genotyping data use DivBrowse [@doi:10.1093/gigascience/giad025] storage engine and BrAPI interface. Endpoints for sample data are implemented using AGENT database SQL to BrAPI broker service.
To integrate those BrAPI endpoint provider into a single service and URL scheme, we work on their integration in a BrAPI proxy service. As next steps, we will expand BrAPI implementation to enable the integration of analysis pipelines in the AGENT portal, e.g. for genebank mining tools such as the FIGS+ pipeline developed by AGENT partner ICARDA [@doi:10.22004/AG.ECON.266624]. Another perspective is to integrate the data collected in the AGENT project into the European Search Catalogue for Plant Genetic Resources (EURISCO) [@doi:10.1093/nar/gkac852] and to implement BrAPI endpoints to make data on PGR collections in European genebanks programmatically accessible.


#### MIAPPE ISA to BrAPI service

Phenotyping is crucial in the breeding process as it enables the identification of desirable traits, selection of breeding lines, and evaluation of breeding success. In the plant community, MIAPPE (Minimal Information About a Plant Phenotyping Experiment) [@doi:10.1111/nph.16544] is the established standard for phenotyping experiments and is commonly serialized as ISA Tab [@doi:10.1038/ng.1054]. Although ISA Tab is easy to read for non-technical experts due to its file-based approach, it lacks programmatic access, particularly for web applications. BrAPI, which is aligned with MIAPPE, can help solve this problem.
MIRA is a tool that enables the automatic deployment of a BrAPI server on a MIAPPE-compliant dataset in ISA Tab format. It can be deployed from a Docker image with the dataset mounted. By utilizing the mapping between MIAPPE, ISA, and BrAPI, there is no need for parsing or manual mapping of datasets that are already compliant with (meta-)data standards. By gaining programmatic access through BrAPI to these datasets, it facilitates the integration of phenotyping datasets into web applications.


#### MIAPPE "BrAPI to ISA" service

Since the release of BrAPI 1.3, efforts have been made to incorporate support for the Minimum Information About Plant Phenotyping Experiments (MIAPPE) standard into the specification [@doi:10.1111/nph.16544]. This integration was finalized in BrAPI 2.0, resulting in full compatibility between the two standards. Consequently, BrAPI now encompasses all attributes necessary for MIAPPE compliance, adhering to standardized descriptions in accordance with MIAPPE guidelines. Leveraging BrAPI as a standardized RESTful web service API specification, we employ the ISA standard for storing metadata and phenotyping data in a standardized manner. This data is structured in the ISA-TAB file format and subjected to validation using the [MIAPPE ISA configuration](https://github.com/ELIXIR-Belgium/isatab-validation). The "BrAPI to ISA" service functions as a converter between BrAPI RESTful endpoints and ISA-TAB, facilitating the archiving of metadata and data and thereby enhancing data preservation and accessibility. The [BrAPI2ISA](https://github.com/elixir-europe/plant-brapi-to-isa) tool is designed to be compatible with BrAPI 1.3, and we invite contributions from the community to extend support for the latest versions of BrAPI.


#### BrAPIMapper

<!-- Valentin G: I'm planning to add a figure in supplementary data that shows data sets on a side, the mapper with some internal parts and the exposed BrAPI services. -->
BrAPIMapper is a full BrAPI implementation of all calls for any data source missing BrAPI implementation or compliance with some BrAPI versions. BrAPIMapper is provided as a docker application that can get its external data sources from mySQL or PostgreSQL databases (with a dedicated interface for Chado database schema), generic REST services (with a dedicated interface for BrAPI endpoints), flat files (XML, JSON, CSV/TSV/GFF3/VCF, YAML) or any combination of any of those. It provides an administration interface to map BrAPI data models to external data sources. The interface allows administrators to select the BrAPI specification versions to use and the calls to enable. Data mapping configuration export and import features simplify upgrades to future BrAPI specifications changes as administrators would only have to map missing fields or make minor adjustments. Amongst others, it supports paging, search calls, either by providing direct results or using deferred results with a search identifier, lists, authentication and manages access restrictions to calls that can be setup through the administration interface as well. This tool aims to accelerate BrAPI services deployment while ensuring specification compliance. 


### Data visualization 
<!-- 
* General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->


#### Flapjack

[Flapjack](https://ics.hutton.ac.uk/flapjack) [@doi:10.1093/bioinformatics/btq580] is a multi-platform desktop application for data visualization and breeding analysis (eg, pedigree verification, marker-assisted backcrossing and forward breeding) using high-throughput genotype data. Data can be easily imported into Flapjack from any BrAPI compatible data source with genotype data available. [Flapjack Bytes](https://github.com/cropgeeks/flapjack-bytes) is a smaller, lightweight and fully web-based counterpart to Flapjack, which can be easily embedded into a database website to provide similar visualizations online. Traditionally supporting its own text-based data formats, Flapjack's use of BrAPI has streamlined the end-user experience for data import and work is underway to determine the best methods to exchange analysis results using future versions of the API.


#### Helium

Helium (<https://helium.hutton.ac.uk>) [@doi:10.1186/1471-2105-15-259] is a plant pedigree visualization platform designed to account for the specific problems that are unique to plant pedigrees. A pedigree is a representation of how genetically discrete individuals are related to one another and is therefore a representation of the genetic relationship between individual plant lines, their parents and progeny. Plant pedigrees are often used to check for potential genotyping or phenotyping errors, since these errors, by the very nature of Mendelian inheritance, are constrained by the pedigree structure in which they exist (Paterson 2011). The accurate representation of pedigrees, and the ability to pull pedigree data from different data sources is therefore important in plant breeding and genetics and therefore ways to visualize and interact this complex data in meaningful ways is critical.

From its original desktop interface (<https://github.com/cardinalb/helium-docs/wiki>), Helium has developed into a web-based visualization platform implementing BrAPI calls to allow users to import data from other BrAPI compliant databases (<https://helium.hutton.ac.uk>). The ability to pull data from BrAPI compliant data sources has significantly expanded Helium’s capability and utility within the community. Helium is used in projects ranging in size from tens to tens of thousands of lines and across a wide variety of crops and species. While originally designed for plant data [@doi:10.3389/fpls.2024.1268847] it has also found utility in other non-plant projects [@doi:10.1007/s10592-024-01611-z] highlighting its broad utility. This also allows Helium users to provide direct dataset links to collaborators allowing the original data to be held with the data provider and utilising Helium for its visualization functionality. Our current Helium deployment includes example BrAPI calls to a barley dataset at Hutton to allow users to test the system and features it offers. 


#### Trait Selector BrAPP

<!-- the "BrAPP" description part should better fit at the beginning of the section. -->
BrAPPs are simple tools developed by the BrAPI community that are entirely reliant on BrAPI for their data requirements. This means a single BrAPI can be shared and used by many organizations, as long as those organizations have the standard BrAPI endpoints available.

The Trait Selector BrAPP is used to search and select useful traits, using a visual aid to help the user find exactly what they need. This BrAPP works with both breeding databases and genebanks. Breeding databases would need to only implement the trait, observation and observation variable calls, while genebanks would require trait, germplasm attribute and germplasm attribute value calls. So, BrAPI servers compliant with version 2 implementing any of these sets of calls would just need to follow the documented steps to create an SVG image of a plant of interest in order to use this BrAPP. CassavaBase and MGIS are two successful examples of the use of this BrAPP. (example screenshots coming + supplementary data: links to the git and the doc)


#### DArTView

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
DArTView is a desktop application for visualizing genotype variant data and looking for trends or correlations. It is newly BrAPI compatible and can use BrAPI as an input data source.


#### DivBrowse

DivBrowse [@doi:10.1093/gigascience/giad025] is a web platform for exploratory data analysis of huge genotyping studies. The software can be run standalone or integrated as a plugin into existing data web portals. It provides a powerful interactive visualization of variant call matrices with hundreds of millions of variants and thousands of samples and enables easy data import and export by using standardized and established bioinformatics file formats. 
At its core, DivBrowse combines the convenience of a genome browser and adds features tailored to the diversity analysis of germplasm. It is able to display genomic features such as nucleotide sequence, associated gene models and short genomic variants. DivBrowse provides visual access to large VCF files obtained through genotyping experiments. In addition to visualizing variant calls per variant and genotype, DivBrowse also calculates and displays variant statistics such as minor allele frequencies, proportion of heterozygous calls or missing variant calls for each visualized genomic window. In addition, dynamic Principal Component Analyses (PCAs) can be performed on a user specified genomic area to provide information on local genomic diversity. 
DivBrowse has a Javascript API to control the tool from a hosting web portal (e.g. to control the list of genotypes to be displayed and the reference genome). DivBrowse has an interface to BLAST, which can be used to directly access genes or other genomic features. The modular structure of DivBrowse also allows developers to configure and easily embed links to external information systems. Furthermore, parts of BrAPI are implemented to provide genotypic data via its server-side component and is also able to consume and visualize genotypic data via an external BrAPI endpoint through the client-side GUI.


### Analytics
<!-- 
* General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->


#### QBMS

Modern breeding programs can utilize data management systems to maintain both phenotypic and genotypic data. Numerous systems are available for adoption. To fully leverage the benefits of digitalization in this ecosystem, breeders need to utilize data from different sources to make efficient data-driven decisions. With increased computational power at their disposal, scientists can construct more advanced analysis pipelines by combining various data sources.

[QBMS](https://icarda-git.github.io/QBMS) [@doi:10.5281/zenodo.10791627] R package eliminates technical barriers scientists experience when using the BrAPI calls in their analysis scripts and pipelines. This barrier arises from the complexity of managing API backend processes, such as authentication, tokens, TCP/IP protocol, JSON format, pagination, stateless calls, asynchronous communication, database IDs, and more. To bridge this gap, we have developed the QBMS R package. This package abstracts the technical complexities, providing breeders (targetted end users) with stateful action verbs/functions familiar to them when navigating their GUI systems. It enables them to query and extract data into a standard data frame structure, consistent with their use of R language, one of the most common statistical tools in the breeding community.

Since its release on the official CRAN repository in October 2021, the QBMS R package has garnered over 9400 downloads. Several tools, such as MrBean, rely on the QBMS package as their source data adapter. Moreover, the community has started building extended solutions on top of it. QBMS can serve as a cornerstone in the breeding modernization revolution by providing access to actionable data and enabling the creation of dashboards to reduce the time between harvest and decision-making for the next breeding cycle.


#### Mr. Bean

Mr.Bean [@doi:10.3389/fpls.2023.1290078] is a graphical user interface designed to assist breeders, statisticians, and individuals involved in plant breeding programs with the analysis of field trials. By utilizing innovative methodologies such as SpATS for modeling spatial trends and autocorrelation models to address spatial variability, Mr.Bean proves highly practical and powerful in facilitating faster and more effective decision-making. Modeling Genotype-by-environment interaction poses its challenges, but Mr.Bean offers the capability to explore various variance-covariance matrices, including Factor Analytic, compound symmetry, and heterogeneous variances, among others, aiding in the assessment of genotype performance across diverse environments.

Mr.Bean boasts flexibility in importing different file types, yet for users managing their data within data management systems (DMS), the process of downloading from their DMS and importing it into MrBean can be cumbersome. To address this issue, QBMS operates in the back-end. This feature prompts users to input the URL of the server, their credentials if necessary, and the specific trial they wish to analyze. Subsequently, users can seamlessly access and utilize their dataset within the entire interface.


#### G-Crunch

G-Crunch is an upcoming user-facing analysis tool that attempts to fill the space of simple, user driven analytics requests, with a generic user interface and the ability to swap out data sources and analysis tools. G-Crunch hopes to streamline repeatable, debuggable simple analytic requests and results.
G-Crunch, as a tool, couldn't feasibly exist without BrAPI. The support of BrAPI interfaces allows G-Crunch to use one unified request method, and adapt to the user's (BrAPI-compliant) existing network of tools, which lowers the barrier to entry for adoption.


### Samples and Genotypes

<!-- * General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->


#### GIGWA

Gigwa is a JEE web application providing means to centralize, share, finely filter, and visualize high-throughput genotyping data [@doi:10.1093/gigascience/giz051]. Built on top of MongoDB, it is scalable and can support working smoothly with datasets containing billions of genotypes. Installable from docker images or all-in-one bundle archives, it is pretty straightforward to deploy on servers or local computers and has thus been adopted by numerous research institutes from around the world. Notably, Gigwa serves as a collaborative management tool and/or a portal for exposing the data for genebanks and breeding programs for some CGIAR centers [@doi:10.1002/ppp3.10187]. Thus, the amount of data hosted and made widely accessible using this system has kept growing over the last few years.

Gigwa developers have been involved in the BrAPI community since 2016 and took part in designing the genotype-related part of the API's specifications. Its first BrAPI-compliant features were designed for compatibility with the Flapjack visualization tool [@doi:10.1093/bioinformatics/btq580] and thus primarily turned it into a BrAPI datasource. Consequently, over time, Gigwa being the first and most reliable application implementing BrAPI-Genotyping server calls, local collaborators and even external partners used it as a reference solution to design a number of tools taking advantage of those features (e.g., [BeegMac](https://webtools.southgreen.fr/BrAPI/Beegmac/), [SnpClust](https://github.com/jframi/snpclust), [QBMS](https://github.com/icarda-git/QBMS)). But further use-cases also required Gigwa to be able to consume data from other BrAPI servers, which led to also implement API-client features into the system. Thanks to all this work, a close collaboration was progressively established with the Integrated Breeding Platform team developing the widely used Breeding Management System, that ended up in both applications now being frequently deployed together, Gigwa pulling germplasm or sample metadata from BMS, and BMS displaying Gigwa-hosted genotypes within its own UI.

Client BrAPI libraries being available for R, community members typically write ad-hoc scripts syndicating data from multiple BrAPI sources (for instance phenotypes from a datasource and genotypes from another) in order to run various kinds of analyses such as GWAS, genomic selection or phylogenetic investigations. As a perspective, we may expect the most generic and widely-used of those pipelines to be at least publicly distributed, and possibly web-interfaced using solutions like R-Shiny in order to provide new, excitingly useful online services, based on Gigwa-hosted data.


#### PHG

<!-- Lynn J. -->
The Practical Haplotype Graph (PHG) is a graph-based computational framework that represents large-scale genetic variation and is optimized for plant breeding and genetics.  Using a pangenome approach, each PHG stores haplotypes (the sequence of part of an individual chromosome) to represent the collected genes of a species.  This allows for a simplified approach for dealing with large scale variation in plant genomes.  The PHG pipeline provides support for a range of genomic analyses and allows for the use of graph data to impute complete genomes from low density sequence or variant data.

Users access the crop databases either with direct calls to the PHG embedded server or indirectly using the rPHG library from an R environment.  The PHG server accepts BrAPI endpoint queries to return information on sample lists and the variants used to define the graph's haplotypes.  In addition, PHG users utilize the BrAPI variantsets endpoint query to return links to VCF files containing haplotype data. Work on the PHG is ongoing.  We expect to support additional BrAPI endpoints that allow for slicing genotypic data based on samples and regions.


### Data Portal

<!-- * General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->


#### FAIDARE

<!-- Cyril P, first draft-->
FAIDARE (<https://urgi.versailles.inrae.fr/faidare/>) is a data discovery portal providing a biologist friendly search system over a global federation of 33 plant research databases. It allows to identify data resources using a full text approach completed with domain specific filters and to link back to the original database for visualization, analysis and download. For instance, it is possible to search for "wheat drought" then to refine the search to the "Triticum aestivum" taxon and yield component traits such as "Thousand Grain Weight". The indexed data types are very broad and include genomic features, such as genes or transposable elements, selected bibliography, QTL, markers, genetic variation studies, phenomic studies and plant genetic resources ie germplasm. This inclusiveness is achieved thanks to a two stage indexation data model. The most generic one provides basic search functionalities and relies on five fields : name, link back URL, data type, species and exhaustive description. The filtering is directly tied to some of those fields. Therefore, to provide more advanced filtering, FAIDARE is also providing a second stage indexation mechanism by taking advantage of BrAPi endpoints to get more detailed metadata on genotyping and phenotyping studies as well as germplasm. In parallel, FAIDARE provides a pre-visualization of germplasm and studies using dedicated cards.
![Figure FAIDARE Federation](images/Schema_FAIDARE.png){#fig:Schema_FAIDARE width="100%"}
The indexation mechanism relies on a dedicated public software (<https://github.com/elixir-europe/plant-brapi-etl-faidare>) that allows data resources manager to request the indexation of there database using pull requests. This BrAPI client is able to extract data from any BrAPI 1.3 and 1.2 endpoint and development of BrAPI 2.x indexation will be initiated in 2025. Since not all databases are willing to implement BrAPI endpoints, we also provide the possibility to generate metadata as BrAPI json files, hence using the standard as a file exchange format.
FAIDARE architecture has been designed by elaborating on the GnpIS Software Architecture [@doi:10.34133/2019/1671403]. As a consequence, BrAPI is at the core of its datamodel, and in particular the JSON data files served by the Elasticsearch NoSQL engine are enriched version of the BrAPI JSON files. FAIDARE also includes a BrAPI endpoint that serves all indexed metadata.
FAIDARE has been adopted by several communities and in particular in the ELIXIR and EMPHASIS european infrastructures. It is also used by the WheatIS of the Wheat-Initiative. Several databases are added each year to the FAIDARE global federation, allowing to increase both the portal and the BrAPI adoption.


#### GLIS

The Global Information System (GLIS) on Plant Genetic Resources for Food and Agriculture (PGRFA) of the International Treaty on Plant Genetic Resources for Food and Agriculture (ITPGRFA) is a web-based global entry point for users and third-party systems to access information and knowledge on scientific, technical and environmental matters to strengthen PGRFA conservation, management and utilization activities. The system and its portal also enable recipients of PGRFA to make available all non-confidential information on germplasm according to the provisions of the Treaty and facilitates access to the results of their research and development. 
 
Thanks to the adoption of Digital Object Identifiers (DOIs) to PGRFA ex situ and in situ based on the Multi-Crop Passport Descriptors (MCPD), the Portal provides access to 1.7 million PGRFA in collections conserved worldwide. Of these, over 1.5 million are accessible for research, training and plant breeding in the food and agriculture domain.
 
The Scientific Advisory Committee of the International Treaty and the Governing Body have repeatedly welcomed efforts on interoperability among germplasm information systems. In this context, the GLIS Portal adopted the Breeding API (BrAPI v1) in 2022. Integrating the BrAPI among the GLIS content negotiators facilitates queries and the exchange of content for data management in plant breeding. The Portal also offers other protocols (XML, DarwinCore, JSON and JSON-LD) to increase data and metadata connectivity. In the near future, depending on the availability of resources, upgrading to BrAPI v2 is planned.


#### FLORILÈGE (Gateway to French Plant Genetic Resources)

Designed primarily for the general public, Florilège provides access to public collections of all French plant biological resources centers. This web portal allows to browse available plant genetic resource accessions and gives the possibility to order some seeds or plant material for cultivation. It includes plant genetic resources of around fifty plant genus from 19 genebanks.

Florilège retrieves accession information from different BrAPI-compliant systems. They include OLGA, a genebank accessions information management system, and GnpIS[@doi:10.34133/2019/1671403] [@doi:10.1007/978-1-4939-6658-5_5], an INRAE data repository for plant genetic resources, phenomics and genetics. Using BrAPI to gather data from these systems reduced the efforts and enabled standardized data retrieval. As a consequence, BrAPI is the de facto standard for exchanging data within the French plant genetic resources community. The Florilège team also requested several update of the BrAPi specifications to better serve this use case, such as Collection or improved external references.
![Figure Florilege Workflow](images/Schema_Florilege.jpg){#fig:Schema_Florilege width="100%"}

Florilège is developed in Drupal 10, and uses xnttbrapi module (to easily connect to BrAPI compliant external databases). 


#### BrAPI plug and play GraphQL based data-warehouse

Using the "Zendro" set of automatic software program-code generators (zendro-dev.github.io) a fully functional, efficient, and cloud-capable BrAPI data-warehouse has been created for the current version of the BrAPI data models. The resulting data-warehouse has two interfaces, one application programming interface implemented in the form of a GraphQL web-server and another intuitive point and click graphical user interface in the browser. Both provide secure access to data read and write functions for all BrAPI data models. These data administration methods comprise create, read, update, and delete (CRUD) functions that are standardized and accept the same parameters for all data models.

While data write access comprises both persisting single or multiple records, data read access is particularly rich in features and includes access to single records referred to by their id and access to multiple records selected by logical filters. In this, multiple records are paginated using the highly efficient cursor based pagination model as proposed in the GraphQL standard. Logical filters allow for exhaustive search queries, whose structure is highly intuitive and based around logical triplets in which a data model field is validated using an operator and a value, e.g. "Study name equals 'xyz'". In this a large collection of operators is available and triplets can be combined to logical search trees using "and" or "or" operators. Searches can be extended over relationships between data models, thus enabling a user to query the warehouse exactly for the data wanted. 

Access security is implemented with the OAuth2 user authentication standard (datatracker.ietf.org/doc/html/rfc6749). Authorization is based on user roles and can be configured differently for each single data model read or write function.

The browser based graphical user interface is implemented in React.js with Next and exposes an intuitive and self explanatory set of functions for each data model. In the left a menu allows the user to access all BrAPI data models. Upon clicking on a model a table is shown which allows the user to paginate through all existing records, sort them by any column, search the records, add new records, or update or delete existing records, if the user role authorizes these functions. Record data can be inspected in a detail view and here relationships to other data records can be reviewed using the very same graphical visual representations. Breadcrumbs allow the user to navigate back and forth in the trail of relationships inspected. Finally, the generated graphical interface allows for the integration of interactive scientific plots and analysis tools written in JavaScript or WebAssembly.

<!-- Asis Hallab: To Do: (i) provide the link to an example running data warehouse, and (ii) include and describe example scientific plots. -->

The Zendro based BrAPI plug and play data-warehouse is capable of forming an efficient cloud of data servers. This is achieved simply by linking (URLs) other Zendro based warehouses that expose the same GraphQL API to the same data models, or a subset of data models. Any network of such Zendro GraphQL servers can be set up using this configuration approach. The code generated then exposes full access to all data records stored on any node of the network, while maintaining full security control at each node. Importantly, the warehouses are programmed in such a way that any number of data servers can be joined without loss of efficiency. Only the network connection speed and size of requested record sets influence the performance.

As explained, Zendro is a code generator and creates a fully functional data warehouse from input data model definitions, i.e. a schema. The schema is given in the form of special data model descriptions, in which each model is defined using JavaScript Object Notation (JSON). Each model is defined in its respective JSON file. A translator has been developed to create the Zendro schema from the BrAPI data model definitions. This ensures that Zendro can create plug and play data warehouses for future versions of the BrAPI with great ease, i.e. by translating the BrAPI models to Zendro input and subsequently running Zendro to create the plug and play warehouse.


## Discussion

<!-- * how can BrAPI help breeders (specifically small breeders)
* Looking ahead - what needs to be done further
* Analytics
* HDP
* GraphQL?
* Field management - treatments etc
* Drones, image processing, HTP
* ChatGPT integration - BreedersGPT
* Weather & soil - why it won’t be in BrAPI -->

### BrAPI for Breeders and Scientists

The BrAPI technical specification document is meant to be read and used by software developers. However, the purpose of the specification, and the community around it, is to make things faster, easier, and cheaper for the breeders and scientists working to make the world a better place. BrAPI offers a convenient path to automation and data integration for software tools in the breeding domain. All of the example use cases described above can be achieved with manual effort, moving and editing data files by hand. However, when the basic structure and flow of data becomes automated, breeders and scientists can spend less time on data management and more time focussing on the science, doing what they do best. For many, the ultimate goal is the development of a digital ecosystem: a collection of software tools and applications that can all work together seamlessly. In this digital ecosystem, data is collected digitally from the beginning, reducing as much human error as possible. The data is checked by quality control and stored automatically, then can be sent to any internal tool or external lab for further analysis with just the click of a button. This idea might sound too good to be true, but as more tools start sharing a universal data standard, automating data flow becomes easier, and the community gets closer to total interoperability. 

### Looking Ahead

The BrAPI specification will continue to grow, enabling more use cases and new types of data. These new use cases might include newer scientific techniques and technologies. Things like drone imaging data, spectroscopy, LIDAR, metabolomics, transcriptomics, high-throughput phenotyping, and machine learning analysis. All of these technologies can open new avenues for research and development of new crop varieties. All of these technologies also generate more data, and require data sharing between different software applications and data repositories. The BrAPI project leadership and community is committed to building the standards to support these new use cases as they arrive and become accepted by the scientific community. In fact, small groups within the BrAPI community have already start building generic data models and communication standards for many of the technologies listed above. These community efforts will eventually become part of the BrAPI standard in a future version of the specification document. 




## Conclusions and Impact

* High level summary of the project/consortium
* BrAPI is fitting into this gap, it doesn’t need to fit these other gaps
* Call to action - Join us!


## References {.page_break_before}

<!-- Explicitly insert bibliography here -->
<div id="refs"></div>

