### Data visualization 
<!-- 
* General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->

#### Flapjack

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
[Flapjack](https://ics.hutton.ac.uk/flapjack) [@doi:10.1093/bioinformatics/btq580] is a multi-platform desktop application for data visualization and breeding analysis (eg, pedigree verification, marker-assisted backcrossing and forward breeding) using high-throughput genotype data. Data can be easily imported into Flapjack from any BrAPI compatible data source with genotype data available. [Flapjack Bytes](https://github.com/cropgeeks/flapjack-bytes) is a smaller, lightweight and fully web-based counterpart to Flapjack, which can be easily embedded into a database website to provide similar visualizations online. Traditionally supporting its own text-based data formats, Flapjack's use of BrAPI has streamlined the end-user experience for data import and work is underway to determine the best methods to exchange analysis results using future versions of the API.

#### Helium

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
Helium (https://helium.hutton.ac.uk) [@doi:10.1186/1471-2105-15-259] is a plant pedigree visualization platform designed to account for the specific problems that are unique to plant pedigrees. A pedigree is a representation of how genetically discrete individuals are related to one another and is therefore a representation of the genetic relationship between individual plant lines, their parents and progeny. Plant pedigrees are often used to check for potential genotyping or phenotyping errors, since these errors, by the very nature of Mendelian inheritance, are constrained by the pedigree structure in which they exist (Paterson 2011). The accurate representation of pedigrees, and the ability to pull pedigree data from different data sources is therefore important in plant breeding and genetics and therefore ways to visualize and interact this complex data in meaningful ways is critical.

From its original desktop interface (https://github.com/cardinalb/helium-docs/wiki), Helium has developed into a web-based visualization platform implementing BrAPI calls to allow users to import data from other BrAPI compliant databases (https://helium.hutton.ac.uk). The ability to pull data from BrAPI compliant data sources has significantly expanded Heliumâ€™s capability and utility within the community. Helium is used in projects ranging in size from tens to tens of thousands of lines and across a wide variety of crops and species. While originally designed for plant data [@doi:10.3389/fpls.2024.1268847] it has also found utility in other non-plant projects [@doi:10.1007/s10592-024-01611-z] highlighting its broad utility. This also allows Helium users to provide direct dataset links to collaborators allowing the original data to be held with the data provider and utilising Helium for its visualization functionality. Our current Helium deployment includes example BrAPI calls to a barley dataset at Hutton to allow users to test the system and features it offers. 




#### Tassel

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
I don't know much about Tassel or its BrAPI compliance. This is filler text for the layout of the manuscript.

#### Trait Selector BrAPP

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
BrAPPs are simple tools developed by the BrAPI community that are entirely reliant on BrAPI for their data requirements. This means a single BrAPI can be shared and used by many organizations, as long as those organizations have the standard BrAPI endpoints available. For example, the Trait Selector BrAPP is used to search and select useful traits, using a visual aid to help the user find exactly what they need. The BrAPP relies only on the BrAPI Trait endpoints to function, so any data resources with a the Trait endpoints available can add this tool to their system.

#### DArTView

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
DArTView is a desktop application for visualizing genotype variant data and looking for trends or correlations. It is newly BrAPI compatible and can use BrAPI as an input data source.

#### DivBrowse

DivBrowse [@doi:10.1093/gigascience/giad025] is a web platform for exploratory data analysis of huge genotyping studies. The software can be run standalone or integrated as a plugin into existing data web portals. It provides a powerful interactive visualization of variant call matrices with hundreds of millions of variants and thousands of samples and enables easy data import and export by using standardized and established bioinformatics file formats. 
At its core, DivBrowse combines the convenience of a genome browser and adds features tailored to the diversity analysis of germplasm. It is able to display genomic features such as nucleotide sequence, associated gene models and short genomic variants. DivBrowse provides visual access to large VCF files obtained through genotyping experiments. In addition to visualizing variant calls per variant and genotype, DivBrowse also calculates and displays variant statistics such as minor allele frequencies, proportion of heterozygous calls or missing variant calls for each visualized genomic window. In addition, dynamic Principal Component Analyses (PCAs) can be performed on a user specified genomic area to provide information on local genomic diversity. 
DivBrowse has a Javascript API to control the tool from a hosting web portal (e.g. to control the list of genotypes to be displayed and the reference genome). DivBrowse has an interface to BLAST, which can be used to directly access genes or other genomic features. The modular structure of DivBrowse also allows developers to configure and easily embed links to external information systems. Furthermore, parts of BrAPI are implemented to provide genotypic data via its server-side component and is also able to consume and visualize genotypic data via an external BrAPI endpoint through the client-side GUI.
