### Data Management 

<!-- * General use case description(s)
* Specific tool examples 
* Alternate solutions/ why is it better with BrAPI
* future related use cases, areas to improve -->

#### PHIS 

The Hybrid Phenotyping Information System ([PHIS](http://www.phis.inrae.fr/)[@doi:https://doi.org/10.1111/nph.15385]), based on the [OpenSILEX](https://github.com/OpenSILEX/) framework, assumes responsibility for the systematic collection and management of data from phenotyping and high-throughput phenotyping experiments on a day-to-day basis. PHIS has the ability to efficiently store, organize and manage a wide range of data sets, including images, spectra and growth curves. This functionality extends to data at multiple spatial and temporal scales, from leaf to canopy, from a variety of sources such as field and greenhouse environments.

A key feature of PHIS is the unambiguous identification of all objects and traits within an experiment, establishing consistent relationships between them through the application of ontologies and semantics. This approach is designed to adapt to variations in experimental conditions, whether in the field or in controlled environments. PHIS's ontology-driven architecture emerges as a robust tool for integrating and managing data derived from diverse experiments and platforms, facilitating the creation of meaningful relationships between objects and augmenting datasets with relevant knowledge and metadata.

Furthermore, PHIS adheres to the Minimal Information About a Plant Phenotyping Experiment ([MIAPPE](https://www.miappe.org/)[@doi:https://doi.org/10.1111/nph.16544]) and the Breeding API (BrAPI) standards. 

The system recommends specific naming conventions, fostering a standardized approach for users to declare their resources. Notably, PHIS is widely adopted by various experimental platforms of the national [PHENOME](https://www.phenome-emphasis.fr/) and European [EMPHASIS](https://emphasis.plant-phenotyping.eu/) infrastructure, serving as a hub for data management. 
Moreover, dedicated instances of PHIS have been established for the explicit purpose of resource sharing, encompassing projects, genetic resources, and variables, thereby fostering collaborative engagement and the dissemination of knowledge pertaining to studied concepts.

##### PHIS & BrAPI

PHIS offers a RESTful API designed to streamline interaction with data within a platform. Within this API, various services aligning with the Breeding API (BrAPI) standards have been implemented, encompassing the Core, Phenotyping, and Germplasm modules. Comprehensive documentation for these services is available on the PHIS Swagger interface. This integration with Swagger ensures that users can easily access, understand, and utilize the functionalities provided by the BrAPI-compliant web services, fostering transparency and facilitating effective engagement with the PHIS platform.

The design and development of PHIS have been meticulously tailored, incorporating the explicit constraint of aligning with BrAPI requirements. This intentional alignment ensures that PHIS adheres to the specified standards and protocols outlined by the Breeding API (BrAPI), thereby fostering seamless integration and compatibility with BrAPI-compliant systems and platforms. This prerequisite served as a substantial foundation for formalizing the data model, simultaneously facilitating compatibility with other standards, such as MIAPPE. Thus, by consciously incorporating BrAPI requirements into its structure, PHIS not only meets the phenotyping domain standards but also enhances its capacity for interoperability and effective collaboration within the broader context of plant breeding and related domains.

The fact that data within a PHIS instance can be queried through BrAPI services makes the indexing of PHIS in [FAIDARE](https://urgi.versailles.inra.fr/faidare/) very easy to implement.

Indeed, as PHIS offers BrAPI-compliant Web Services, this greatly simplifies the integration and data exchange with other European information systems that handle phenotyping data. The adherence to BrAPI standards ensures a common interface and compatibility, facilitating seamless communication and collaboration between PHIS and other systems within the European context. This interoperability not only streamlines data sharing but also promotes a more cohesive and effective approach to managing and utilizing phenotyping data across diverse platforms and research initiatives in the European scientific community.

#### DeltaBreed

<!-- Shawn Y.-->
DeltaBreed is an open-source data management system designed and developed by Breeding Insight to support USDA-ARS specialty crop and animal breeders. DeltaBreed is a unified system for managing breeding data that connects a variety of BrAPI applications (see list below). BrAPI integration allows the complexity underlying interoperability to be hidden, shielding users from multifactorial differences between diverse applications. DeltaBreed, adhering to the BrAPI model, establishes data standards and validations for users and provides a singular framework for data management and user training.

DeltaBreed users need not be aware of BrAPI or the specifics of underlying applications but will notice that BrAPI interoperability reduces the need for human-mediated file transfers and data manipulation. Field Book users, for example, can connect to their DeltaBreed program, authenticate, and pull studies and traits directly from DeltaBreed to Field Book on their data collection device. The subsequent step of pushing observations from Field Book to DeltaBreed is straightforward via BrAPI, but will not be implemented until repeated observation handling workflows are established to differentiate and validate repeated observations, such as accidental repeats, overwrite requests, time-series observations, and repeated sub-entity measures. Users can expect DeltaBreed observation handling to become more seamless with future development.

**DeltaBreed Connected Applications**
<< Submission is expected April 2024. We may need to trim this aspirational list down to reality in final edits.>>

+ BIMS <https://www.breedwithbims.org/>
+ BrAPI Java Server <https://test-server.brapi.org/brapi/v2/>
+ BrAPI Sync <https://github.com/IntegratedBreedingPlatform/brapi-sync>
+ BreedBase <https://breedbase.org/>
+ Diversity Arrays Technologies (DArT) genotyping services
+ Field Book <https://play.google.com/store/apps/details?id=com.fieldbook.tracker>
+ Gigwa <https://gigwa.southgreen.fr/gigwa/>
+ Mr Bean <https://github.com/AparicioJohan/MrBeanApp>
+ Pedigree Viewer <https://github.com/solgenomics/BrAPI-Pedigree-Viewer>

#### BMS

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->
[brapi sync](https://github.com/IntegratedBreedingPlatform/brapi-sync) and other things

#### Breedbase

Breedbase is a comprehensive breeding data management system [@doi:10.1093/g3journal/jkac078][@doi:10.1371/journal.pone.0240059] that implements a digital ecosystem for all breeding data, including trial data, phenotypic data, and genotypic data. Data acquisition is through tabled-based apps such as Fieldbook [@doi:10.2135/cropsci2013.08.0579] and related apps, such as Coordinate and InterCross apps, through drone imagery, Near Infra-Red Spectroscopy (NIRS), and other technologies. Search functions such as the Search Wizard interface provide powerful query capabilities, and various breeding-centric analysis tools are available, including mixed models, heritability, stability, PCA, and various clustering algorithms. The original impetus for creating Breedbase was the advent of new breeding paradigms based on genomic information such as genomic prediction algorithms [@doi:10.1093/genetics/157.4.1819] and the accompanying data management challenges, and complete genomic prediction workflow is integrated in the system.  The first instance was created for the NextGen Cassava project in 2012 as the Cassavabase (https://cassavabase.org/) database. Databases for other CGIAR root, tuber and banana (RTB) crops followed with database for yam (https://yambase.org/), sweet potato (https://sweetpotatobase.org/), banana (https://musabase.org/) as well as instances in labs and companies. The BrAPI interface[@doi:10.1093/bioinformatics/btz190] is crucial for Breedbase: Breedbase communicates via BrAPI with the data collection tablets, connection to other projects such as CLIMMOB [@doi:10.1016/j.compag.2023.108539], and many native tools use the BrAPI interface for accessing data. Users also appreciate the ability to connect to Breedbase instances using packages such as QBMS <https://icarda-git.github.io/QBMS/> for data import into R for custom analyses. Breedbase has been an early and continuous adopter of, and contributor to, the BrAPI standard.

#### BIMS

BIMS (Breeding Information Management System) [@doi:10.1093/database/baab054] is a free, secure, and online breeding management system which allows breeders to store, manage, archive, and analyze their private breeding program data. BIMS enables individual breeders to have complete control of their own breeding data along with access to tools such as data import/export, data analysis and data archiving for their germplasm, phenotype, genotype, and image data. BIMS is currently implemented in five community databases, the Genome Database for Rosaceae [@doi:10.1093/nar/gky1000], CottonGEN [@doi:10.3390/plants10122805], the Citrus Genome Database, the Pulse Crop Database, and the Genome Database for Vaccinium, as well as a crop-independent website, <https://breedwithbims.org>. BIMS in these five community databases enables individual breeders to import publicly available data so that they can utilize public data in their breeding program. BIMS utilizes the Android App Field Book, enabling seamless data transfer between BIMS and the Field Book App through either files or BrAPI. Data transfer through BrAPI between BIMS and other resources such as BreedBase, GIGWA, and Breeder Genomics Hub is also on the way. 

#### Germinate

<!-- Peter S: Stub paragraph to stimulate the writing process. Please edit, rewrite, or delete as needed. -->

[Germinate](https://ics.hutton.ac.uk/get-germinate/) [@doi:10.1002/csc2.20248] is an open-source plant genetic resources database that combines and integrates various kinds of plant breeding data including genotypic data, phenotypic trials data, passport data, images, geographic information and climate data into a single repository. Germinate is tightly linked to the BrAPI specification and supports a majority of BrAPI endpoints for querying, filtering and submission.

Germinate integrates and connects with other BrAPI-enabled tools such as GridScore for phenotypic data collection, Flapjack for genotypic data visualization and Helium for pedigree visualization, but, due to the nature of BrAPI, Germinate can act as a data repository for any BrAPI-compatible tool. Thanks to the interoperability provided by BrAPI the need for manual data handling becomes a rarity with the direct benefit of faster data processing, fewer to no human errors, data security and integrity.
